ros/src/computing/perception/semantics/packages/costmap_generator/test/src/test_costmap_generator.cpp

```diff
@@ -0,0 +1,499 @@
+
+
+  const double expected_x_ind = 2;
+  const double expected_y_ind = 5;
+  EXPECT_EQ(expected_x_ind, index.x());
+  EXPECT_EQ(expected_y_ind, index.y());
+}
+
+TEST_F(TestSuite, CheckValidIndex)
+{
+  grid_map::GridMap costmap = test_obj_.makeDummyCostmap();
+
+  grid_map::Index index(2,5);
+  bool is_valid = test_obj_.isValidInd(costmap, index);
+
+  bool expected_valid = true;
+  EXPECT_EQ(expected_valid, is_valid);
+}
+
+TEST_F(TestSuite, CheckNonValidIndex)
+{
+  grid_map::GridMap costmap = test_obj_.makeDummyCostmap();
+
+  grid_map::Index index(2,10);
+  bool is_valid = test_obj_.isValidInd(costmap, index);
+
+  bool expected_valid = false;
+  EXPECT_EQ(expected_valid, is_valid);
+}
+
+TEST_F(TestSuite, CheckCalculationPointsCostmap)
+{
+  grid_map::GridMap costmap = test_obj_.makeDummyCostmap();
+
+  std::vector<std::vector<std::vector<double>>>  vec_x_y_z = test_obj_.makeDummy3DVec();
+  vec_x_y_z[5][5].push_back(2.2);
+
+  double maximum_height_thres = 3.0;
+  double minimum_lidar_height_thres = -2.0;
+  double grid_min_value = 0;
+  double grid_max_value = 1;
+  std::string layer_name = "test";
+
+  grid_map::Matrix costmap_mat = test_obj_.calculateCostmap(maximum_height_thres, minimum_lidar_height_thres, grid_min_value,
+                                              grid_max_value, costmap, layer_name, vec_x_y_z);
+  double expected_cost = 1.0;
+  EXPECT_DOUBLE_EQ(expected_cost, costmap_mat(5,5));
+}
+
+TEST_F(TestSuite, CheckHeightThresholdForCost)
+{
+  grid_map::GridMap costmap = test_obj_.makeDummyCostmap();
+
+  std::vector<std::vector<std::vector<double>>>  vec_x_y_z = test_obj_.makeDummy3DVec();
+  vec_x_y_z[5][5].push_back(3.2);
+
+  double maximum_height_thres = 3.0;
+  double minimum_lidar_height_thres = -2.0;
+  double grid_min_value = 0;
+  double grid_max_value = 1;
+  std::string layer_name = "test";
+
+  grid_map::Matrix costmap_mat = test_obj_.calculateCostmap(maximum_height_thres, minimum_lidar_height_thres, grid_min_value,
+                                              grid_max_value, costmap, layer_name, vec_x_y_z);
+  double expected_cost = 0.0;
+  EXPECT_DOUBLE_EQ(expected_cost, costmap_mat(5,5));
+}
+
+TEST_F(TestSuite, CheckMakeExpandedPoints)
+{
+  geometry_msgs::Point in_centroid;
+  geometry_msgs::Point32 in_corner_point;
+  double expand_polygon_size = 1;
+
+  in_centroid.x = 0;
+  in_centroid.y = 0;
+  in_centroid.z = 0;
+
+  in_corner_point.x = 1;
+  in_corner_point.y = 1;
+  in_corner_point.z = 1;
+
+  geometry_msgs::Point expanded_point = test_obj_.makeExpandedPoint(in_centroid, in_corner_point, expand_polygon_size);
+  double expected_x = 1.707109;
+  double expected_y = 1.707109;
+  const double buffer = 0.01;
+  EXPECT_NEAR(expected_x, expanded_point.x, buffer);
+  EXPECT_NEAR(expected_y, expanded_point.y, buffer);
+}
+
+TEST_F(TestSuite, ChecMakePolygonFromObjectConvexhull)
+{
+  double expand_polygon_size = 0;
+  autoware_msgs::DetectedObject in_object;
```

Comment:
@cirpue49 `autoware_msgs::DetectedObject in_object;` is used in multiple tests, although its `convex_hull` member varies you can declare it `in_object` as part of  the TestSuite and initialise `header.frame_id` and `pose` in the SetUp as it seems to be the same and clear the `convex_hull` contents in TearDown so that it is fresh for each test

Reply:
Could you check the latest commit?
Added some scripts in SetUp function.